src/field/vecops.cxx

```diff
@@ -195,39 +204,45 @@ const Field2D Div(const Vector2D &v, const Field2D &f, CELL_LOC outloc) {
   result += FDDZ(metric->J * vcn.z, f, outloc);
   result /= metric->J;
 
+  result.setLocation(outloc);
```

Comment:
This is not necessary, result.location is set already by FDDX. In the Field*D difops this is just skipped, but we could put an `ASSERT1(result.getLocation()==outloc)` if you want to double-check?

Reply:
Ah yes the return is a Field not a vector -- I was busy adding these for the routines which return vectors and must have got carried away!

Reply:
Actually (at least in this branch) FDDX (and indexFDDX) on field2D doesn't set the location as far as I can tell.

Reply:
@ZedThree, @johnomotani  should I fix this missing `setLocation` here or in a separate PR?

Reply:
Should note the 2D/3D DDX versions shouldn't need the location setting in vecops either as these are both set in either `indexDDX` and/or `applyXDiff`. It's just `indexFDDX(Field2D)` that doesn't set a location currently.